libyul/optimiser/StackLimitEvader.cpp

```diff
@@ -66,12 +67,37 @@ struct MemoryOffsetAllocator
 		if (unreachableVariables.count(_function))
 		{
 			yulAssert(!slotAllocations.count(_function), "");
+
+			if (functionDefinitions.count(_function))
+			{
+				FunctionDefinition const* functionDefinition = functionDefinitions.at(_function);
+				yulAssert(functionDefinition, "");
+				size_t totalArgCount = functionDefinition->returnVariables.size() + functionDefinition->parameters.size();
+				size_t slotsNeeded = (totalArgCount > 16) ? totalArgCount - 16 : 0;
+
+				if (slotsNeeded)
+					for (TypedName const& param: functionDefinition->parameters)
+					{
+						slotAllocations[param.name] = requiredSlots++;
+						if (!--slotsNeeded)
+							break;
```

Comment:
Yes: moving function arguments can be done locally in the beginning of the respective function body, so that's nicer.
Moving return variables requires modifying the call sites or even generating auxiliary functions.